src/makeInfinityAdder.js

```diff
@@ -4,7 +4,23 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+
+  const adder = (a) => {
+    if (typeof a === 'undefined') {
+      const returnedValue = sum;
+
+      sum = 0;
+
+      return returnedValue;
+    }
+
+    sum += a;
```

Comment:
To ensure that the function only adds numbers, you should validate that 'a' is a number. Otherwise, the function might exhibit unexpected behavior when non-number values are passed.